code/src/components/GamePage.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { DuckContainer, Button } from 'styled-components/GlobalStyles';
+import { generateMoves } from 'reducer/game';
+import styled from 'styled-components';
+import { DucklingHeader } from './DucklingHeader';
+import { EndingPage } from './EndingPage';
+
+export const GamePage = () => {
+  // Getting description and actions from currentLocation in the store
+  const { description, actions } = useSelector((store) => store.game.currentLocation)
+  const dispatch = useDispatch();
+
+  const username = useSelector((store) => store.game.username);
+
+  return (
+    <section>
+      <DuckContainer>
+        <DucklingHeader />
+        <HeaderText>
+          {actions.length > 0 ? (
+            <p>We don&apos;t have
+          a lot of time. Hurry up, {username}!
+            </p>
+          ) : (<p>Thank you for your help, {username}. You&apos;re truly my saviour</p>)}
```

Comment:
We really like you using the username further on in the game. Good for the user experience!